### ISSUE TITLE
Function Module - Identifier must be lower case(keyword_case) and DATAEND issue

### ISSUE THREAD

**Issue by habrahams** [abaplint/abaplint#1231]
I have an issue where a FM name is flagged as it must be lowercase which is never the case. All FM names are always uppercase regardless. I do not know why a DATAEND issue is picked up.
![image](https://user-images.githubusercontent.com/16045067/91283952-938a3480-e78b-11ea-9ae8-77eb9791fc9b.png)

Confirmed in vscode extension and abaplint cli.

**Comment by larshp** [comment]
in this example: https://github.com/larshp/abapGit/blob/master/src/zabapgit_parallel.fugr.z_abapgit_serialize_parallel.abap the developer has pretty printed the function module source code

The "Expected" errors are reported at the wrong positions, in the following example, the BEGIN OF DATA contains something wrong. As WRITE is not allowed inside BEGIN OF DATA, https://syntax.abaplint.org/?filter=data#/structure/Data

For your example, try copying the code into the https://playground.abaplint.org, and delete code until the statement causing the trouble is found

![image](https://user-images.githubusercontent.com/5888506/91284831-b8cb7280-e78c-11ea-9729-ddc2d61ecd67.png)

**Comment by habrahams** [comment]
It looks like the "VALID BETWEEN begda AND endda" is causing the DATAEND issue:
```
  DATA BEGIN OF lt_p0045 OCCURS 0.
          INCLUDE STRUCTURE p0045.
  DATA END OF lt_p0045 VALID BETWEEN begda AND endda.
```

I also reported the "VALID BETWEEN begda AND endda" issue in #1234

**Comment by habrahams** [comment]
w.r.t the FM name that should be lowercase as per check keyword_case, if you pretty print a FM in ADT or in SAP GUI the signature is changed to the example you provided(https://github.com/larshp/abapGit/blob/master/src/zabapgit_parallel.fugr.z_abapgit_serialize_parallel.abap ), but once you save the FM the ABAP backend automatically changes the FM name and signature to uppercase.

for example a FM pretty printed prior to saving:
![image](https://user-images.githubusercontent.com/16045067/91289031-0696a980-e792-11ea-9150-143258674498.png)
and after saving:
![image](https://user-images.githubusercontent.com/16045067/91289087-2201b480-e792-11ea-8511-a7399d8fc8ca.png)

Should it not always be assumed that FM names and signature will alway be uppercase regardless of whether the code was pretty printed or not?

**Comment by larshp** [comment]
try setting `"ignoreFunctionModuleName": true,` in the `keyword_case` rule, https://rules.abaplint.org/keyword_case/

**Comment by habrahams** [comment]
@larshp , thanks, I should have gone and looked at rule first (Note to self)

**Comment by larshp** [comment]
I also did not remember it was there 🤷‍♂️
